src/openms_gui/include/OpenMS/VISUAL/LayerData.h

```diff
@@ -507,8 +517,71 @@ namespace OpenMS
     ExperimentType::SpectrumType cached_spectrum_;
   };
 
+  class LogWindow;
+
+  /// A base class to annotate layers of specific types with (identification) data
+  class LayerAnnotatorBase
+  {
+    public:
+      /**
+        @brief C'tor with params
+        
+        @param supported_types Which identification data types are allowed to be opened by the user in annotate()
+        @param file_dialog_text The header text of the file dialog shown to the user
+      **/
+      LayerAnnotatorBase(const FileTypes::FileTypeList& supported_types, const String& file_dialog_text);
+      
+      /// Annotates a @p layer, writing messages to @p log and showing QMessageBoxes on errors.
+      /// The input file is selected via a file-dialog which is opened with @p current_path as initial path.
+      /// The filetype is checked to be one of the supported_types_ before the annotateWorker_ function is called
+      /// as implemented by the derived classes
+      bool annotate(LayerData& layer, LogWindow* const log, const String& current_path) const;
+
+    protected:
+      /// abstract virtual worker function to annotate a layer using content from the @p filename
+      /// returns true on success
+      virtual bool annotateWorker_(LayerData& layer, const String& filename, LogWindow* const log) const = 0;
```

Comment:
const LogWindow* ?

Reply:
ehh.. no. We'd like to write to the LogWindow.
So the pointer is everything that can be const here.
To avoid confusion we could also go for `LogWindow&`